### PR TITLE
Guide: Steps to Register and Verify an Event Hook

### PR DESCRIPTION
**Pull request by motikagan-okta**
## Description:
- **What's changed?** Adding new Guide content on the process for registering and Verifying an Event Hook.
- **Is this PR related to a Monolith release?** No.

### Resolves:

* [OKTA-231340](https://oktainc.atlassian.net/browse/OKTA-231340)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/implement-your-service/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: Implement Your Service
+---
+
+You need to implement a web service with an Internet-accessible endpoint to receive event hook calls from Okta. It's your responsibility to develop the code and to arrange its hosting on a system external to Okta. Okta defines the REST API contract for the REST requests it sends to your service.
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/implement-your-service/index.md]: I think this first sentence is missing something along the lines of "This guide is intended for developers who need to..."

Comment (jakubvul) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/implement-your-service/index.md]: This first sentence is a bit jarring right now. Maybe add a few more words such as "Before you start..." or "This guide assumes that you are interested in..." or something along those lines. 

Also "Event Hook" should be capitalized.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+title: Register Your Endpoint
+---
+
+After implementing your external service, you need to register it with Okta. To do this, you need to make a REST API call to the [Event Hooks Management API](/docs/reference/api/event-hooks/). You can use a tool like Postman to make the call, and Okta provides a [Postman collection](/docs/reference/api/event-hooks/#getting-started) to make this easier.
+
+You make a `POST` call to `https://{yourOktaDomain}/api/v1/eventHooks` to register your endpoint. The body of the request needs to contain a JSON payload containing an [eventHook](/docs/reference/api/event-hooks/#event-hook-object) object, which specifies the properties of the Event Hook you are registering, including:
+
+ - the URI of the your external service's endpoint
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md]: ~the~

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ - the list of specific event types this event hook should deliver
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md]: Event Hook

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/verify-your-endpoint/index.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+title: Verify Your Endpoint
+---
+
+After registering the event hook, you need to trigger a one-time verification process, by making a `POST` request to `https://{yourOktaDomain}/api/v1/eventHooks/${id}/lifecycle/verify`. The `id` value here is the system-generated unique ID value for the Event Hook object, which Okta returned in the `eventHook` object when you registered the event hook.
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/verify-your-endpoint/index.md]: Event Hook x 2

And I'd shorten this to just say:

"The `id` value here belongs to the Event Hook object which Okta returned when..."

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/implement-your-service/index.md (hunk 1 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/implement-your-service/index.md]: Changed to "Before you start....". Corrected capitalization of Event Hook.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md (hunk 2 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md]: Thanks, corrected.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md (hunk 3 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/register-your-endpoint/index.md]: Corrected.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/verify-your-endpoint/index.md (hunk 4 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/verify-your-endpoint/index.md]: Made suggested edit.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+title: Ongoing Delivery of Events
+---
+
+After your event hook is verified, delivery of events to your endpoint can begin. There can be a delay, however, of up to a minute between successful verification and the start of event delivery.
```
Comment (amoghvasekar-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md]: nit : Maybe `There can be a delay, however, of up to a minute between successful verification and the start of event delivery.` is more relevant after the next line which explains when an event hook is fired?
The 1 minute delay is related to `Every time an event of a type that the Event Hook is subscribed to occurs in your Okta org...` part.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/overview/index.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+title: Overview
+---
+
+Event Hooks are outbound calls from Okta that can be used to notify your own software systems of events occurring in your Okta org. See [Event Hooks](/docs/concepts/event-hooks/) for an overview. This guide provides the steps for setting up an Event Hook in your Okta org, which are:
+
+1. Implement your external web service to receive Event Hook calls from Okta.
```
Comment (amoghvasekar-okta) [inline packages/@okta/vuepress-site/docs/guides/set-up-event-hook/overview/index.md]: Are these numbers all expected to be `1.`?

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/overview/index.md]: Yes, it's just markdown notation for a numbered list. Will get rendered as sequence of numbers.

File: packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md (hunk 5 above)
Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md]: Have removed the statement. Need to follow up because something similar was stated in the Event Hooks conceptual overview page.

Comment (amoghvasekar-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md]: Sorry for the confusion. Let me clarify - 
the statement is correct

I just meant maybe it should be moved down a little bit in terms of structure.
The way it is right now, it could also be interpret to mean `Once I verify hook, delivery takes up to a minute to start`. While it actually is - `Once I verify a hook, it takes up to a minute for Okta to know eligible events are to be delivered to this hook`. That is, an event fired immediately after verification may not be recognized by Okta as an event to deliver.

I guess its too subtle. I'll let you decide if the current wording / structure is fine. But we should definitely have it in.

Comment (motikagan-okta) [reply packages/@okta/vuepress-site/docs/guides/set-up-event-hook/ongoing-delivery-of-events/index.md]: Thanks Amogh, that makes sense. I've updated to: "There can be a delay, however, of up to a minute between successful verification of an Event Hook and it becoming operational."
